src/components/table/table_root.tsx

```diff
@@ -28,30 +31,33 @@ import { useTableColumns } from './use_table_columns.js';
 import { useTableScroll } from './use_table_scroll.js';
 
 const CustomHTMLTable = styled(HTMLTable, {
-  shouldForwardProp: (prop) => prop !== 'striped' && prop !== 'stickyHeader',
-})<{ stickyHeader: boolean }>`
+  shouldForwardProp: (prop) =>
+    prop !== 'striped' && prop !== 'stickyHeader' && prop !== 'noHeader',
+})<{ stickyHeader: boolean; noHeader: boolean }>`
   /* When using a sticky header, ensure that the borders are located below the last header instead of above the first row. */
-  ${(props) => {
-    if (!props.stickyHeader) return '';
-
-    return `
+  ${(props) =>
+    match(props)
+      .with({ stickyHeader: false, noHeader: false }, () => '')
+      .otherwise(
+        (props) => `
       thead tr:last-child {
         box-shadow: inset 0 -1px #11141826;
       }
     
       tbody tr:first-of-type td {
-        box-shadow: ${
-          props.bordered
-            ? 'inset 1px 0 0 0 #11141826 !important'
-            : 'none !important'
-        };
+        box-shadow: ${match(props)
+          .with(
+            { bordered: true },
+            () => 'inset 1px 0 0 0 #11141826 !important',
+          )
+          .otherwise(() => 'none !important')};
```

Comment:
Its a bit strange for me to use ts-pattern only for a ternary

Reply:
Initially thought I needed more than 2 cases.

But not addressing this comment as I think it's a matter of personal preferences.